src/sidebar/util/fetch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * An error indicating a failed network request.
+ *
+ * Failures that this error can represent include:
+ *
+ *  - Failures to send an HTTP request
+ *  - Requests that returned non-2xx responses
+ *  - Failures to parse the response in the expected format (eg. JSON)
+ */
+export class FetchError extends Error {
```

Comment:
I chose the `FetchError` name to match the `fetchJSON` operation. `FetchJSONError` seemed a bit unwieldy and not appropriate if we wanted to reuse this for requests that return non-JSON responses.  This error intentionally covers a variety of errors (networking - where there is no response, failure response, failure to parse response) so I thought it better to name the error after the operation rather than the kind of error.